### PR TITLE
fix: redrafter sampling

### PR DESCRIPTION
**Pull request by 1ytic**
Fix issue #3125 about incorrect sampling from Redrafter model. Original Apple implementation from [ml-recurrent-drafter](https://github.com/apple/ml-recurrent-drafter/blob/bd8586bb9bbfa761644c36f932a0cd1eb3f1cdf9/recurrent_drafting/modeling_drafter.py#L201) uses normalized draft logits, while current TensorRT-LLM implementation uses unnormalized. As a result, the validation and rejection sampling are incorrect [here](https://github.com/1ytic/TensorRT-LLM/blob/385a01055cbecafd450d242e25cd6ac9dcb9537a/tensorrt_llm/models/redrafter/redrafter_helper.py#L87) and [here](https://github.com/1ytic/TensorRT-LLM/blob/385a01055cbecafd450d242e25cd6ac9dcb9537a/tensorrt_llm/models/redrafter/redrafter_helper.py#L689).

### CODE REVIEW

**Comment by juney-nvidia** [comment]
@rakib-hasan  Hi Rakib, can you help review this MR?

Thanks
June

**Comment by rakib-hasan** [comment]
Thanks for fixing the issue, @1ytic .
Could you please rename the variables as well so that they represent what is being stored? Like:
```
    log_p_token_in_beam = None
...
...
        cur_log_p_token_in_beam = unsqueeze(
            _gather_beams(log_p_new_token, gather_indices, batch_size,
                          num_beams), 2)  # [bs, nb, 1, V]
        if log_p_token_in_beam is None:  # first iteration
            log_p_token_in_beam = cur_log_p_token_in_beam
        else:
            log_p_token_in_beam = concat(
                [
                    _gather_beams(log_p_token_in_beam, gather_indices,
                                  batch_size,
                                  num_beams),  # prev_top_log_p [bs, nb, i, V]
                    cur_log_p_token_in_beam
                ],
                dim=2)  # [bs, nb, i+1, V]
        last_tokens = top_token_ids  # [bs, nb]
    return beams, log_p_token_in_beam
```

**Comment by 1ytic** [comment]
Done. Should I remove `_beam_search_candidates_v0` function as well?

**Comment by rakib-hasan** [comment]
> Should I remove _beam_search_candidates_v0 function as well?

Sure. I think we kept it long enough.

**Review by rakib-hasan — APPROVED** [review]
Thanks. LGTM.

**Comment by rakib-hasan** [comment]
@1ytic can you fix the formatting issue that caused the CI to fail?

**Comment by 1ytic** [comment]
> @1ytic can you fix the formatting issue that caused the CI to fail?

Done. Sorry about this. Should I rerun bot?

**Comment by rakib-hasan** [comment]
No worries. Yes, CI needs to pass before it can be merged.

**Comment by symphonylyh** [comment]
release check failed due to formatting issue @1ytic

**Comment by 1ytic** [comment]
> release check failed due to formatting issue @1ytic

I don't think I can fix this issue with yapf.

**Comment by symphonylyh** [comment]
> > release check failed due to formatting issue @1ytic
> 
> I don't think I can fix this issue with yapf.

is it an intermittent issue or from a broken main? I just triggered a rebase and re-run check

**Comment by 1ytic** [comment]
> > > release check failed due to formatting issue @1ytic
> > 
> > 
> > I don't think I can fix this issue with yapf.
> 
> is it an intermittent issue or from a broken main? I just triggered a rebase and re-run check

I checked this logs [Release Check Pipeline #302](https://nv/trt-llm-cicd/job/helpers/job/ReleaseCheck_PR/302/) and it looks strange to me. This fix is quite simple, it shouldn't trigger this issue.

**Comment by rakib-hasan** [comment]
Earlier CI passed as well. I don't understand why yapf failed this time.
